data/src/main/java/com/microsoft/azure/kusto/data/auth/ConnectionStringBuilder.java

```diff
@@ -54,14 +61,53 @@ private ConnectionStringBuilder(String clusterUrl) {
         this.userNameForTracing = null;
         this.clientVersionForTracing = null;
         this.applicationNameForTracing = null;
-        this.initProcessNameForTracing();
+        this.InitTracingParams();
     }
 
-    private void initProcessNameForTracing() {
+    private void InitTracingParams() {
         // sun.java.command holds the cmd line used to invoke the running application
         this.processNameForTracing = UriUtils.stripFileNameFromCommandLine(System.getProperty("sun.java.command"));
         // user.name is used by jvm to hold the user name
-        this.userNameForTracing = System.getProperty("user.name");
+        this.userNameForTracing = getOsUser();
+
+        this.sdkVersion = buildHeaderFormat(
+                Pair.of("Kusto.Java.Client", Utils.getPackageVersion()),
+                Pair.of(getRuntime(), getVersion()));
+    }
+
+    private static String getOsUser() {
+        String user = System.getProperty("user.name");
+        if (StringUtils.isBlank(user)) {
+            user = System.getenv("USERNAME");
+            String domain = System.getenv("USERDOMAIN");
+            if (StringUtils.isNotBlank(domain) && StringUtils.isNotBlank(user)) {
+                user = domain + "\\" + user;
+            }
+        }
+        return StringUtils.isNotBlank(user) ? user : "[none]";
+    }
+
+    private static String getVersion() {
+        String version = System.getProperty("java.version");
+        if (StringUtils.isBlank(version)) {
+            return "UnknownVersion";
+        }
+        return version;
+    }
+
+    private static String getRuntime() {
```

Comment:
Can you explain why did you choose this prioritization ? java.runtime.name is expected to always return openJDK .. so no added info here 
although i dont think any of this matters - vendor is the only one that varys

Reply:
Not true, it can return for example "Java(TM) 2 Runtime Environment, Standard Edition", OpenJDK is returned only when used with openJDK..